sway-ir/src/optimize/dce.rs

```diff
@@ -58,6 +58,13 @@ fn is_removable_store(
         InstOp::MemCopyBytes { dst_val_ptr, .. }
         | InstOp::MemCopyVal { dst_val_ptr, .. }
         | InstOp::Store { dst_val_ptr, .. } => {
+            // if the pointer comes from unknown sources, it is unsafe to delete the store
```

Comment:
As discussed, this special case check can now be removed. The handling of `Incomplete` `get_reffered_symbols` result below treats this and other escapes.

Reply:
Done.